### PR TITLE
Remove DNS resolution for IPv6 route

### PR DESCRIPTION
**Pull request by hidrarga**
## Problem

Sometimes when the VPN client is restarting, the DNS resolution doesn't work anymore (although it works when the VPN is down). I don't know if that's a bug from OpenVPN or if we are missing something in the firewall rules or network configuration.

## Solution

I'm using the environment variables provided by OpenVPN : https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/#environmental-variables

With this solution, we don't need to store IPv6 of the VPN server in a config variable, which makes the client connection much faster.

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by hidrarga** [comment]
It takes less than 30s on a cube with this method (it was taking more than 60s with DNS resolution issues)

**Comment by hidrarga** [comment]
!testme

**Comment by hidrarga** [comment]
I've added some improvements for the firewall rules: when the VPN client is connecting, it provides some environmental variables, such as `trusted_ip` OR `trusted_ip6` variables, but not both at the same time, so I can only create a rule for IPv4 or IPv6... Actually, I don't know if that's a problem?

Otherwise, I could use the variables `ifconfig_ipv6_remote` and `route_vpn_gateway` but I'm not sure if they are valid for all clients?

What do you think of this PR @zamentur @alexAubin ?

**Comment by hidrarga** [comment]
!testme

File: conf/scripts/route-down.d/40-vpnclient-unset-ipv6 [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
 is_ip6addr_set() {
   local ip6_addr=${1}
-  ip address show dev tun0 2> /dev/null | grep -q "${ip6_addr}/"
+  ip address show dev "${gateway_interface}" 2> /dev/null | grep -q "${ip6_addr}/"
 }
 
 ip6_addr=$(yunohost app setting "vpnclient" "ip6_addr")
+gateway_interface=${dev}
```
Comment (alexAubin) [inline conf/scripts/route-down.d/40-vpnclient-unset-ipv6:9]: Eeeh not sure to understand where `$dev` is coming from ? Is this coupled to the way openvpn invoke the script ?

**Comment by alexAubin** [comment]
(Ah yes that's listed in the doc you mention : https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/#environmental-variables )

File: conf/scripts/route-down.d/40-vpnclient-unset-ipv6 (hunk 1 above)
Comment (alexAubin) [reply conf/scripts/route-down.d/40-vpnclient-unset-ipv6:9]:
```suggestion
# cf https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/#environmental-variables for where 'dev' comes from
gateway_interface=${dev}
```

File: conf/scripts/route-up.d/40-vpnclient-set-ipv6 [hunk 2]
```diff
@@ -2,13 +2,15 @@
 
 is_ip6addr_set() {
   local ip6_addr=${1}
-  ip address show dev tun0 2> /dev/null | grep -q "${ip6_addr}/"
+  ip address show dev "${gateway_interface}" 2> /dev/null | grep -q "${ip6_addr}/"
 }
 
 ip6_addr=$(yunohost app setting "vpnclient" "ip6_addr")
+gateway_interface=${dev}
```
Comment (alexAubin) [inline conf/scripts/route-up.d/40-vpnclient-set-ipv6:9]:
```suggestion
# cf https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/#environmental-variables for where 'dev' comes from
gateway_interface=${dev}
```

**Pull request merged**